### PR TITLE
Dev/architecture v11 comparator fiellds search redo

### PR DESCRIPTION
**Pull request by TonyKatkov89**

### CODE REVIEW

File: hypex/executor/executor.py [hunk 1]
```diff
@@ -150,6 +150,34 @@ def _field_searching(
...
+            if isinstance(role, AdditionalRole):
+                for column in columns:
+                    searched_data = searched_data.add_column(
+                        data=data.additional_fields[column], role={column: role}
+                    )
+            else:
+                for column in columns:
+                    searched_data = searched_data.add_column(
+                        data=data.ds[column], role={column: role}
+                    )
```
Comment (VasilievaKA) [inline hypex/executor/executor.py:178]: You can replace this: 
```
for column in columns:
  t_data = data.additional_fields[column] if isinstance(role, AdditionalRole) else data.ds[column] 
searched_data = searched_data.add_column(data=t_data, role={column: role})
````

**Review by VasilievaKA — APPROVED** [review]
Attractive!

File: hypex/executor/executor.py (hunk 1 above)
Comment (Dmatryus) [reply hypex/executor/executor.py:178]:
```py
for column in columns:
  t_data = data.additional_fields[column] if isinstance(role, AdditionalRole) else data.ds[column] 
searched_data = searched_data.add_column(data=t_data, role={column: role})
```

File: hypex/comparators/abstract.py [hunk 2]
```diff
@@ -225,118 +210,88 @@ def _split_for_groups_mode(
...
         compare_by: Literal["groups", "columns", "columns_in_groups", "cross"],
-        target_fields: Union[str, List[str]],
-        baseline_field: Optional[str] = None,
-        group_field: Optional[Union[str, Dataset]] = None,
-        additional_fields: Optional[Dataset] = None,
-    ) -> Tuple:
+        target_fields_data: Dataset,
+        baseline_field_data: Dataset,
+        group_field_data: Dataset,
+    ) -> Tuple[List[Tuple[str, Dataset]], List[Tuple[str, Dataset]]]:
```
Comment (Dmatryus) [inline hypex/comparators/abstract.py:290]: Wrap into a specific type

File: hypex/executor/executor.py [hunk 3]
```diff
@@ -150,6 +150,34 @@ def _field_searching(
             )
         return searched_field
 
+    @staticmethod
+    def _field_data_search(
```
Comment (Dmatryus) [inline hypex/executor/executor.py:154]: Move to ExperimentData

**Review by Dmatryus — CHANGES_REQUESTED** [review]
It's excellent, but move to ExperimentData 👍

File: hypex/executor/executor.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/executor/executor.py]: done

File: hypex/comparators/abstract.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/comparators/abstract.py:289]: done

**Review by Dmatryus — APPROVED** [review]

**Pull request merged**